pkg/apis/scorecard/v1alpha1/formatter.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/mattn/go-isatty"
+)
+
+const (
+	redColor   = "31"
+	greenColor = "32"
+	noColor    = "%s\n"
+)
+
+func (s Test) MarshalText() (string, error) {
+	var sb strings.Builder
+
+	failColor := ": \033[1;" + redColor + "m%s\033[0m\n"
+	passColor := ": \033[1;" + greenColor + "m%s\033[0m\n"
+
+	// turn off colorization if not in a terminal
+	if !isatty.IsTerminal(os.Stdout.Fd()) &&
+		!isatty.IsCygwinTerminal(os.Stdout.Fd()) {
+		passColor = noColor
+		failColor = noColor
+	}
+
+	sb.WriteString(fmt.Sprintf("%s:\n", s.Spec.Labels["suite"]))
+	sb.WriteString("\tLabels: \n")
+	for labelKey, labelValue := range s.Spec.Labels {
+		sb.WriteString(fmt.Sprintf("\t\t%q:%q\n", labelKey, labelValue))
+	}
+	for _, result := range s.Status.Results {
+		sb.WriteString(fmt.Sprintf("\t%-35s ", result.Name))
+		if result.State == PassState {
+			sb.WriteString(fmt.Sprintf(passColor, PassState))
+		} else if result.State == FailState {
```

Comment:
What about error state?

Reply:
true, with scorecard2, the error state can be returned, so it would probably be presented in the same color as a fail...so if its a 'pass' its green else its 'red'.